src/services/ghost/GhostStreamingParser.ts

```diff
@@ -0,0 +1,510 @@
+			return text
+				.replace(/\r\n/g, "\n") // Normalize line endings
+				.replace(/\r/g, "\n") // Handle old Mac line endings
+				.replace(/\t/g, "    ") // Convert tabs to spaces
+				.replace(/[ \t]+$/gm, "") // Remove trailing whitespace from each line
+		}
+
+		const normalizedContent = normalizeWhitespace(content)
+		const normalizedSearch = normalizeWhitespace(searchPattern)
+
+		// Try normalized match
+		index = normalizedContent.indexOf(normalizedSearch)
+		if (index !== -1) {
+			// Map back to original content position
+			return this.mapNormalizedToOriginalIndex(content, normalizedContent, index)
+		}
+
+		// Try trimmed search (remove leading/trailing whitespace)
+		const trimmedSearch = searchPattern.trim()
+		if (trimmedSearch !== searchPattern) {
+			index = content.indexOf(trimmedSearch)
+			if (index !== -1) {
+				return index
+			}
+		}
+
+		return -1 // No match found
+	}
+
+	/**
+	 * Map an index from normalized content back to the original content
+	 */
+	private mapNormalizedToOriginalIndex(
+		originalContent: string,
+		normalizedContent: string,
+		normalizedIndex: number,
+	): number {
+		let originalIndex = 0
+		let normalizedPos = 0
+
+		while (normalizedPos < normalizedIndex && originalIndex < originalContent.length) {
+			const originalChar = originalContent[originalIndex]
+			const normalizedChar = normalizedContent[normalizedPos]
+
+			if (originalChar === normalizedChar) {
+				originalIndex++
+				normalizedPos++
+			} else {
+				// Handle whitespace normalization differences
+				if (/\s/.test(originalChar)) {
+					originalIndex++
+					// Skip ahead in original until we find non-whitespace or match normalized
+					while (originalIndex < originalContent.length && /\s/.test(originalContent[originalIndex])) {
+						originalIndex++
+					}
+					if (normalizedPos < normalizedContent.length && /\s/.test(normalizedChar)) {
+						normalizedPos++
+					}
+				} else {
+					// Characters don't match, this shouldn't happen with proper normalization
+					originalIndex++
+					normalizedPos++
+				}
+			}
+		}
+
+		return originalIndex
+	}
+
+	/**
+	 * Extract cursor position from content containing cursor marker
+	 */
+	private extractCursorPosition(content: string): number | undefined {
+		const markerIndex = content.indexOf(CURSOR_MARKER)
+		return markerIndex !== -1 ? markerIndex : undefined
+	}
+
+	/**
+	 * Remove cursor marker from content and return clean content
+	 */
+	private removeCursorMarker(content: string): string {
+		return content.replaceAll(CURSOR_MARKER, "")
+	}
+
+	/**
+	 * Get the current buffer content (for debugging)
+	 */
+	public getBuffer(): string {
+		return this.buffer
+	}
+
+	/**
+	 * Get completed changes (for debugging)
+	 */
+	public getCompletedChanges(): ParsedChange[] {
+		return [...this.completedChanges]
+	}
+
+	/**
+	 * Conservative XML sanitization - only fixes the specific case from user feedback
+	 */
+	private sanitizeXMLConservative(buffer: string): string {
```

Comment:
I resisted adding "performance" comments to this function- (use a regex here, blah blah). Good job making it readable! I guess the suggestions will never be _that_ long, so we shouldn't have to worry about performance here?